obspy/io/kml/core.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+"""
+Keyhole Markup Language (KML) output support in ObsPy
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+from math import log
+
+from lxml.etree import Element, SubElement, tostring
+
+
+def inventory_to_kml_string(inventory):
+    """
+    Convert an :class:`~obspy.core.inventory.inventory.Inventory` to a kml
+    string representation.
+
+    :type inventory: :class:`~obspy.core.inventory.inventory.Inventory`
+    :param inventory: Input station metadata.
+    :rtype: str
+    :return: String containing KML information of the station metadata.
+    """
+    res = ""
+    return res
+
+
+def catalog_to_kml_string(
+        catalog,
+        icon_url="http://maps.google.com/mapfiles/kml/shapes/earthquake.png",
+        label_func=None, icon_size_func=None):
+    """
+    Convert an :class:`~obspy.core.event.Catalog` to a kml string
+    representation.
+
+    :type catalog: :class:`~obspy.core.event.Catalog`
+    :param catalog: Input catalog data.
+    :type icon_url: str
+    :param icon_url: Internet URL of icon to use for events (e.g. png image).
+    :type label_func: func
+    :type label_func: Custom function to use for determining each event's
+        label. User provided function is supposed to take an
+        :class:`~obspy.core.event.Event` object as single argument, e.g. for
+        empty labels use `label_func=lambda x: ""`.
+    :type icon_size_func_func: func
+    :type icon_size_func_func: Custom function to use for determining each
+        event's icon size. User provided function is supposed to take an
+        :class:`~obspy.core.event.Event` object as single argument.
+    :rtype: str
```

Comment:
Are you sure this is not `bytes`? Since lxml prefers to parse `bytes` and determine encoding from the XML, I'd assume it produces `bytes` as well.